bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlAccessHandler.java

```diff
@@ -0,0 +1,300 @@
+                    "@text/offline.configuration-error.deviceId");
+            return;
+        }
+
+        nhcAccess.setEventHandler(this);
+
+        updateProperties(nhcAccess);
+
+        String location = nhcAccess.getLocation();
+        if (thing.getLocation() == null) {
+            thing.setLocation(location);
+        }
+
+        this.nhcAccess = nhcAccess;
+
+        initialized = true;
+        deviceInitialized();
+    }
+
+    @Override
+    void refresh() {
+        NhcAccess access = nhcAccess;
+        if (access != null) {
+            accessBellEvent(access.getBellState());
+            accessDoorLockEvent(access.getDoorLockState());
+            if (access.getType().equals(AccessType.RINGANDCOMEIN)) {
+                accessRingAndComeInEvent(access.getRingAndComeInState());
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        NikoHomeControlCommunication nhcComm = getCommunication(getBridgeHandler());
+        if (nhcComm != null) {
+            NhcAccess access = nhcComm.getAccessDevices().get(deviceId);
+            if (access != null) {
+                access.unsetEventHandler();
+            }
+        }
+        nhcAccess = null;
+        super.dispose();
+    }
+
+    private void updateProperties(NhcAccess nhcAccess) {
+        Map<String, String> properties = new HashMap<>();
+
+        if (nhcAccess instanceof NhcAccess2) {
+            NhcAccess2 access = (NhcAccess2) nhcAccess;
+            properties.put(PROPERTY_DEVICE_TYPE, access.getDeviceType());
+            properties.put(PROPERTY_DEVICE_TECHNOLOGY, access.getDeviceTechnology());
+            properties.put(PROPERTY_DEVICE_MODEL, access.getDeviceModel());
+        }
+
+        String buttonId = nhcAccess.getButtonId();
+        if (buttonId != null) {
+            properties.put("buttonId", buttonId);
+        }
+
+        if (nhcAccess.supportsVideoStream()) {
+            properties.put("username", "admin");
+            properties.put("password", "123qwe");
+            String ipAddress = nhcAccess.getIpAddress();
+            if (ipAddress != null) {
+                properties.put("ipAddress", ipAddress);
+            }
+            String mjpegUri = nhcAccess.getMjpegUri();
+            if (mjpegUri != null) {
+                properties.put("mjpegUri", mjpegUri);
+            }
+            String tnUri = nhcAccess.getTnUri();
+            if (tnUri != null) {
+                properties.put("tnUri", tnUri);
+            }
+        }
+
+        thing.setProperties(properties);
+    }
+
+    @Override
+    public void accessBellEvent(boolean state) {
+        NhcAccess nhcAccess = this.nhcAccess;
+        if (nhcAccess == null) {
+            logger.debug(" access device with ID {} not initialized", deviceId);
+            return;
+        }
+
+        updateState(CHANNEL_BELL_BUTTON, state ? OnOffType.ON : OnOffType.OFF);
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
This can be triggered by an event in the connected system. In a number of cases, the OH thing remained offline, but events started comming in through the communications layer. If this happens, the thing is effectively online and can be switched online.

Reply:
If you are sure it is necessary, that is fine for me.